devito/tools/data_structures.py

```diff
@@ -480,3 +481,75 @@ def __hash__(self):
                 h ^= hash((key, value))
             self._hash = h
         return self._hash
+
+
+class UnboundedMultiTuple(object):
+
+    """
+    An UnboundedMultiTuple is an ordered collection of tuples that can be
+    infinitely iterated over.
+
+    Examples
+    --------
+    >>> ub = UnboundedMultiTuple([1, 2], [3, 4])
+    >>> ub
+    UnboundedMultiTuple((1, 2), (3, 4))
+    >>> ub.iter()
+    >>> ub
+    UnboundedMultiTuple((1, 2)*, (3, 4))
```

Comment:
nitpicking: shouldn't the `*` be before the tuple? THis looks like the tip passed the first tuple

Reply:
yes, fixing